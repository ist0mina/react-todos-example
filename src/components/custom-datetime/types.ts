export interface CustomDatetimeProps {
    controlId: string;
    title?: string;
    selected?: Date;
    disabled?: boolean;    
    onChange?: (date: Date | null) => void;    
};
