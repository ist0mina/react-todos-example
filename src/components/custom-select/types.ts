export type OptionSelect = {
    key: string;
    value: string;
};

export interface CustomSelectProps {
    controlId: string;
    title?: string;
    data?: OptionSelect[];
    selected?: string;
    disabled?: boolean;
    onChange?: (event: React.ChangeEvent<HTMLSelectElement>) => void;
};