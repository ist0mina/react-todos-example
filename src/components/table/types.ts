export interface TableBasicProps {
    hasNNCol?: boolean;
    hasControlCol?: boolean;
}

export type TableColumItem = {
    title: string;
}

export interface TableHeaderProps extends TableBasicProps {
    columns: TableColumItem[];
}

export type TableCellItem = {
    text: string;
}

export type TableRowItem = {
    cells: TableCellItem[];
}

export interface TableBodyProps extends TableBasicProps {
    rows: TableRowItem[]; 
}

export interface TableRowProps extends TableBasicProps {
    cols: TableCellItem[];
    index: number;
}

export type SimpleTableProps = TableHeaderProps & TableBodyProps;