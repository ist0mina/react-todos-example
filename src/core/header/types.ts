export interface HeaderStateProps {    
    pathname: string;
};

export type HeaderProps = HeaderStateProps;