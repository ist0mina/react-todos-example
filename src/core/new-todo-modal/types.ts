import { ToggleFieldType } from '../../features/toggle/types';
import { TodoItem } from '../../features/todos/types';

export interface NewTodoModalStateProps {
    show?: boolean;
    maxId?: number;
}

export interface NewTodoModalDispatchProps {
    toggle?: (field: ToggleFieldType) => void;
    addTodo?: (todo: TodoItem) => void;
}

export interface NewTodoModalOwnProps {
    handleClose: () => void;
    handleOk: (text: string) => void;
}

export type NewTodoModalProps = NewTodoModalStateProps & NewTodoModalDispatchProps & NewTodoModalOwnProps;