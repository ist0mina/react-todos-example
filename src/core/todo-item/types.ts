import { TodoStatusType } from '../../features/todos/types';

export interface TodoItemStateProps {    
    text: string;
    open: boolean;
    status: TodoStatusType;
    dateCreated: Date;
    dateLastChanged?: Date;
};

export interface TodoItemDispatchProps {
    toggle: (todo: number) => void;
    openTodo: (id: number) => void;
}

export interface TodoItemOwnProps {
    id: number;
};

export type TodoItemProps = TodoItemStateProps & TodoItemDispatchProps & TodoItemOwnProps;