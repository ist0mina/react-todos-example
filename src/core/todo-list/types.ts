export interface TodoListStateProps {
    ids: number[];
};

export type TodoListProps = TodoListStateProps;