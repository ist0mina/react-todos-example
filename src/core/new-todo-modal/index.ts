export * from './NewTodoModal';
export * from './NewTodoModalContainer';
export * from './types';