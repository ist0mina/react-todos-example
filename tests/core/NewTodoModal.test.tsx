import { makeMountrender, reduxify, snapshotify, mockStore } from '../utils';

import { NewTodoModalContainer } from '../../src/core/new-todo-modal';

import { ToggleActionType, ToggleFieldType } from '../../src/features/toggle/types';

describe('<NewTodoModalContainer>', () => {
    it('matches snapshots', () => {        
        const store = mockStore();
        store.dispatch({ type: ToggleActionType.TOGGLE, payload: ToggleFieldType.NEW_TODO_MODAL });

        const component = reduxify({ Component: NewTodoModalContainer, store });
        const wrapper = makeMountrender(component)();        
        
        console.log(wrapper.find('NewTodoModal'));

        expect(snapshotify(wrapper)).toMatchSnapshot();
    });

    it('modal closed', () => {
        const component = reduxify({ Component: NewTodoModalContainer });
        const wrapper = makeMountrender(component)();        

        console.log(wrapper.html());
    });
});