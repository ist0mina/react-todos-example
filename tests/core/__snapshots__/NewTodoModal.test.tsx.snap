// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewTodoModalContainer> matches snapshots 1`] = `"<div class=\\"fade modal-backdrop show\\"></div><div role=\\"dialog\\" aria-modal=\\"true\\" class=\\"fade modal show\\" tabindex=\\"-1\\" style=\\"display: block;\\"><div role=\\"document\\" class=\\"modal-dialog\\"><div class=\\"modal-content\\"><div class=\\"modal-header\\"><div class=\\"modal-title h4\\">New todo</div><button type=\\"button\\" class=\\"close\\"><span aria-hidden=\\"true\\">Ã—</span><span class=\\"sr-only\\">Close</span></button></div><form novalidate=\\"\\" class=\\"\\"><div class=\\"modal-body\\"><div class=\\"form-group\\"><label class=\\"form-label\\" for=\\"formNewTodoText\\">Todo text</label><input required=\\"\\" placeholder=\\"Todo text...\\" id=\\"formNewTodoText\\" class=\\"form-control\\" value=\\"\\"><div class=\\"invalid-feedback\\">Need enter new todo</div></div></div><div class=\\"modal-footer\\"><button type=\\"button\\" class=\\"btn btn-outline-dark\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"ban\\" class=\\"svg-inline--fa fa-ban fa-w-16 mr-1\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path fill=\\"currentColor\\" d=\\"M256 8C119.034 8 8 119.033 8 256s111.034 248 248 248 248-111.034 248-248S392.967 8 256 8zm130.108 117.892c65.448 65.448 70 165.481 20.677 235.637L150.47 105.216c70.204-49.356 170.226-44.735 235.638 20.676zM125.892 386.108c-65.448-65.448-70-165.481-20.677-235.637L361.53 406.784c-70.203 49.356-170.226 44.736-235.638-20.676z\\"></path></svg>Cancel</button><button type=\\"submit\\" class=\\"btn btn-primary\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"check\\" class=\\"svg-inline--fa fa-check fa-w-16 mr-1\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path fill=\\"currentColor\\" d=\\"M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z\\"></path></svg>Ok</button></div></form></div></div></div>"`;
